db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/remove_from_provider.rb

```diff
@@ -13,7 +13,7 @@
   case removal_type
   when "remove_from_disk"
     $evm.log('info', "Removing VM:<#{vm.name}> from provider:<#{ems.try(:name)}>")
-    vm.remove_from_disk(false)
+    vm.remove_from_disk(false) if ems
```

Comment:
@durandom 
We've had a few scenarios recently where retirement has had issues completing retirement successfully after it's restarted due to provider issues.

Changing this method to proceed if there's no ems will take care of multiple issues for us:
I don't think it's too much logic for the state machine. We have to consider many different use cases. In situations like this and others like it, the VM has already been removed from the provider and finishing the retirement process without error allows for the VM to be removed from the VMDB and email notifications to be sent.  
  
We have a Trello card setup to fix this method:
https://trello.com/c/vuk9G1wi  

If you want, you could remove your change to the remove_from_provider method since I'll be changing it soon.

Please let me know if you have any questions or would like to discuss it further.